template/table.py

```diff
@@ -8,31 +8,50 @@
 TIMESTAMP_COLUMN = 2
 SCHEMA_ENCODING_COLUMN = 3
 
+
 class Base_page:
-        # Array of Pages num_columns long plus schema and indirections columns
-        # size() - 2 = schema column;  size() - 1 indirection column
-        def __init__(self, num_columns):
-            self.columns = [Page() for i in range(num_columns + 2)] 
+    '''
+    :param tail_page_list: list      Stores a list of all Tail_Page objects associated with the Base_page
+    :param columns_list: list        Stores a list of all Page objects for the Base_page, these are the data columns
+    :param tail_page_count: int      An integer that holds the current amount of tail pages the Base_page has
+    '''
+    def __init__(self, num_columns: int):
+        # Create a starting Tail Page for updates
+        self.tail_page_list = [Tail_page(num_columns)]
+        # Create a list of Physical Pages num_columns long plus Indirection, RID, TimeStamp, and Schema columns
```

Comment:
This looks great Jim! What are you thinking in terms of the RID for the tail pages? Meaning how will the base pages correspond to the indexes of the tail pages in the tail_page_list? (When you want to make an update to a base page, where does the indirection pointer tell you to go?)

Reply:
So that's a good question, with this implementation, the potential to have this RID dict in the indirection column of the base/tail page is there, but that may be a little too multipurpose. So, I propose this and let me know what you think:
* A dict that is used only for the indirection column and has context scoped to the Base_page meaning it knows what index of the list of Tail_pages the base page has, and the index within that page. This would allow us to trim the tail_page keys out of the RID as it exists now, and I think make that logic simpler since every base page has it's specific tail pages tied to it.

Reply:
Im going to merge to master so Brittany and I can write the pseudocode for query. I am not sure I follow what you're saying above so maybe we can hop on a call this afternoon or you can upload a picture.